src/libopenrave-core/colladaparser/colladareader.cpp

```diff
@@ -967,6 +977,12 @@ class ColladaReader : public daeErrorHandler
                 break;
             }
         }
+
+        if( bSuccess ) {
+            // for extracting a single body from file, need to assign a random body id to not conflict with what's already in environment
+            pbody->_id = str(boost::format("body%d")%utils::GetNanoTime());
```

Comment:
@ziyan same as above

Reply:
For now, I think empty id is okay for this case. Resolved.